python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -370,21 +438,106 @@ def fit(self, X, y):
 
         return self
 
-    def predict(self, X):
+    def predict(self, X, deep_check=False):
         """
         Predicts the labels for X.
 
+        Parameters
+        ----------
+        X : dask cuDF dataframe or numpy array (n_samples, n_features)
+        deep_check : boolean (default = False)
+                     This is used to check if the concatenated forest used to
+                     predict the labels in GPU, have the right forest
+                     information.
+                     Set it to True if you want to run an extensive check of
+                     the concatenated treelite forest created. It will compare
+                     the position and value of each node present in each tree
+                     of the concatenated forest with respect to the original
+                     forests present in the different workers.
+                     Required only while using predict for binary
+                     classification.
+        Returns
+        ----------
+        y: dask cuDF or numpy array of shape (n_samples, 1)
+
+        """
+        for w in self.workers:
+            if self.rfs[w].result()._multi_class:
+                self._multi_class = True
+
+        if self._multi_class:
```

Comment:
fp64 also forces cpu prediction right? also, probably should allow users to pass in an option to get CPU prediction.